lib/matplotlib/dates.py

```diff
@@ -732,20 +733,105 @@ def __call__(self, x, pos=None):
 
 
 class rrulewrapper(object):
+    def __init__(self, freq, tzinfo=None, **kwargs):
+        kwargs['freq'] = freq
```

Comment:
I actually have no clue if it is always safe to modify a `**kwargs`. Does it never modify a passed in dictionary?

Reply:
I have looked into this a bit and I don't see anyone actually *saying* this is safe, but it also doesn't seem like the question has come up before - I would guess that it would be surprising if it *were* unsafe and someone would have commented on it by now. My understanding of how these things are supposed to work makes me fairly certain that this will not be a problem, but I've [opened a question on SO](https://stackoverflow.com/questions/45883655/is-it-always-safe-to-modify-the-kwargs-dictionary) just in case.

Either way, if we find that there's some weird situation where it could be a problem, we can just have a second PR that adds `kwargs = kwargs.copy()` at the top of the function, so I don't see any reason why uncertainty in this regard should be blocking.